server/src/main/java/org/elasticsearch/action/admin/indices/stats/CommonStats.java

```diff
@@ -255,147 +325,147 @@ public int hashCode() {
             requestCache,
             recoveryStats,
             bulk,
-            shards
+            shards,
+            nodeMappings
         );
     }
 
     public void add(CommonStats stats) {
-        if (docs == null) {
-            if (stats.getDocs() != null) {
+        if (stats.getDocs() != null) {
```

Comment:
Why make this more complicated? We seem to have the null checks in the add methods so aren't they redundant here?

Reply:
Not in all of them. E.g., org.elasticsearch.index.cache.query.QueryCacheStats#add doesn't have the check. Neither org.elasticsearch.index.fielddata.FieldDataStats#add nor org.elasticsearch.index.cache.request.RequestCacheStats#add . So I was bumping into NPE in the tests. I will add the checks also to their add methods now that I realized. But I believe we should leave the new approach here for safety -- you never know if a new Stats comes by and someone forgets to add the check. And since the number of `if` statements is the same, I think this is safer. Right?

Reply:
@original-brownbear , I'm guessing the comment above was left over from last week. Because I did revert a lot of these. Feel free to check again.